src/main/java/com/karankumar/bookproject/ui/goal/ComponentGoalEvent.java

```diff
@@ -0,0 +1,19 @@
+package com.karankumar.bookproject.ui.goal;
+
+import com.karankumar.bookproject.backend.entity.ReadingGoal;
+import com.vaadin.flow.component.ComponentEvent;
+
+public class ComponentGoalEvent {
+    public abstract static class GoalFormEvent extends ComponentEvent<ReadingGoalForm> {
+    private final ReadingGoal readingGoal;
+
+    protected GoalFormEvent(ReadingGoalForm source, ReadingGoal readingGoal) {
+        super(source, false);
+        this.readingGoal = readingGoal;
+    }
+
+    public ReadingGoal getReadingGoal() {
+        return readingGoal;
+    }
```

Comment:
This section should be indented four space

Reply:
@knjk04 every line shoud have 4 spaces?

Reply:
@mareksmelko777 Every line should be indented four spaces from the previous indentation level. 

Here, you've indented the `GoalFormEvent` class line. That's fine. However, the body of the inner class should also be indented.